packages/lexical-selection/src/range-selection.ts

```diff
@@ -416,6 +417,27 @@ export function $wrapNodesImpl(
   }
 }
 
+/**
+ * Tests if the selection's parent element has vertical writing mode.
+ * @param selection - The selection whose parent to test.
+ * @returns true if the selection's parent has vertical writing mode (writing-mode: vertical-rl), false otherwise.
+ */
+export function $isEditorVerticalOrientation(
+  selection: RangeSelection,
+): boolean {
+  const anchorNode = selection.anchor.getNode();
+  const parent = $isRootNode(anchorNode)
+    ? anchorNode
+    : anchorNode.getParentOrThrow();
+  const editor = $getEditor();
+  const domElement = editor.getElementByKey(parent.getKey());
+  if (domElement === null) {
+    return false;
+  }
+  const computedStyle = window.getComputedStyle(domElement);
```

Comment:
When iframes are in use, the editor's window may not match the global window

```suggestion
  const computedStyle = getWindow(editor).getComputedStyle(domElement);
```

Could also use `domElement.ownerDocument.defaultView` instead of `getWindow(editor)`, should be the same thing at least under the current constraints of how lexical DOM is allowed to work

Reply:
Thanks for the suggestion! I've pushed a [commit](https://github.com/facebook/lexical/pull/7439/commits/147dcf3fe3377b75e0b9f9b151398a3daf63da5f) using `domElement.ownerDocument.defaultView`.